client/src/app/service/auth.service.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpClient, HttpParams } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { map } from 'rxjs/operators';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { environment } from 'src/environments/environment';
+
+export interface User {
+  username: string;
+  token?: string;
+}
+
+@Injectable({
+  providedIn: 'root'
+})
+export class AuthService {
+  private currentUserSubject: BehaviorSubject<User>;
+
+  constructor(private http: HttpClient) {
+    const token: string | null = localStorage.getItem('userToken');
+    const username: string | null = localStorage.getItem('username');
+    if(token) {
+      const user = {
+        username: username,
+        token: token
+      } as User
+      this.currentUserSubject = new BehaviorSubject<User>(user);
+    }
+    else {
+      this.currentUserSubject = new BehaviorSubject<User>({} as User)
+    }
+  }
+
+  public get currentUserValue(): User {
+    return this.currentUserSubject.value;
+  }
+
+  // tslint:disable-next-line: typedef
+  login(username: string, password: string) {
```

Comment:
I also like and recommends explicitly adding all access modifiers and returned types, it makes code much cleaner.
Also why are you disabling tslint typedef here?